hhvm-pgsql doesn't build against hhvm 3.8.0

Here's the first of a handful of seemingly related errors:

```

Scanning dependencies of target pgsql
[ 25%] Building CXX object CMakeFiles/pgsql.dir/pgsql.cpp.o
/home/rrh/git-work-2/php_agent/hhvm-pgsql/pgsql.cpp: In static member function ‘static HPHP::{anonymous}::PGSQL* HPHP::{anonymous}::PGSQL::Get(const HPHP::Variant&)’:
/home/rrh/git-work-2/php_agent/hhvm-pgsql/pgsql.cpp:210:67: error: cannot convert ‘HPHP::SmartPtr<HPHP::{anonymous}::PGSQL>’ to ‘HPHP::{anonymous}::PGSQL*’ in initialization
     PGSQL *pgsql = conn_id.toResource().getTyped<PGSQL>(true, true);
                                                                   ^

```

--- a/pdo_pgsql.cpp
+++ b/pdo_pgsql.cpp
@@ -12,9 +12,14 @@ PDOPgSql::PDOPgSql() : PDODriver("pgsql") {
 }
 
 SmartPtr<PDOResource> PDOPgSql::createResourceImpl() {
-        return makeSmartPtr<PDOPgSqlResource>(std::make_shared<PDOPgSqlConnection>());
+    return makeSmartPtr<PDOPgSqlResource>(std::make_shared<PDOPgSqlConnection>());
 }
 
+SmartPtr<PDOResource> PDOPgSql::createResourceImpl(const sp_PDOConnection& conn) {
+    return makeSmartPtr<PDOPgSqlResource>(
+        std::dynamic_pointer_cast<PDOPgSqlConnection>(conn)
+    );
+}
 
 long pdo_attr_lval(const Array& options, int opt, long defaultValue){
     if(options.exists(opt)){

--- a/pdo_pgsql.h
+++ b/pdo_pgsql.h
@@ -8,6 +8,7 @@ namespace HPHP {
 struct PDOPgSql : public PDODriver {
     PDOPgSql();
     virtual SmartPtr<PDOResource> createResourceImpl() override;
+    virtual SmartPtr<PDOResource> createResourceImpl(const sp_PDOConnection& conn) override;
 };
 
 long pdo_attr_lval(const Array& options, int opt, long defaultValue);

--- a/pdo_pgsql_statement.cpp
+++ b/pdo_pgsql_statement.cpp
@@ -200,7 +200,7 @@ stmt_retry:
             }
         }
 
-        PDOColumn *col = columns[colno].toResource().getTyped<PDOColumn>();
+        PDOColumn *col = columns[colno].toResource().getTyped<PDOColumn>().get();
         col->name = String(m_result.fieldName(colno));
         col->maxlen = m_result.size(colno);
         col->precision = m_result.precision(colno);
@@ -322,7 +322,7 @@ stmt_retry:
 
         char* val = m_result.getValue(current_row, colno);
 
-        PDOColumn* col = columns[colno].toResource().getTyped<PDOColumn>();
+        PDOColumn* col = columns[colno].toResource().getTyped<PDOColumn>().get();
 
         switch(col->param_type){
             case PDO_PARAM_INT:

--- a/pgsql.cpp
+++ b/pgsql.cpp
@@ -207,7 +207,7 @@ PGSQL *PGSQL::Get(const Variant& conn_id) {
         return nullptr;
     }
 
-    PGSQL *pgsql = conn_id.toResource().getTyped<PGSQL>(true, true);
+    PGSQL *pgsql = conn_id.toResource().getTyped<PGSQL>(true, true).get();
     return pgsql;
 }
 
@@ -303,7 +303,7 @@ PGSQLResult *PGSQLResult::Get(const Variant& result) {
         return nullptr;
     }
 
-    auto *res = result.toResource().getTyped<PGSQLResult>(true, true);
+    auto *res = result.toResource().getTyped<PGSQLResult>(true, true).get();
     return res;
 }
 
@@ -1656,12 +1656,12 @@ public:
     {
         Hdf pgsql = hdf["PGSQL"];
 
-        PGSQL::AllowPersistent     = Config::GetBool(ini, pgsql["AllowPersistent"], true);
-        PGSQL::MaxPersistent       = Config::GetInt32(ini, pgsql["MaxPersistent"], -1);
-        PGSQL::MaxLinks            = Config::GetInt32(ini, pgsql["MaxLinks"], -1);
-        PGSQL::AutoResetPersistent = Config::GetBool(ini, pgsql["AutoResetPersistent"]);
-        PGSQL::IgnoreNotice        = Config::GetBool(ini, pgsql["IgnoreNotice"]);
-        PGSQL::LogNotice           = Config::GetBool(ini, pgsql["LogNotice"]);
+        PGSQL::AllowPersistent     = Config::GetBool(ini,  pgsql, "AllowPersistent", true);
+        PGSQL::MaxPersistent       = Config::GetInt32(ini, pgsql, "MaxPersistent", -1);
+        PGSQL::MaxLinks            = Config::GetInt32(ini, pgsql, "MaxLinks", -1);
+        PGSQL::AutoResetPersistent = Config::GetBool(ini,  pgsql, "AutoResetPersistent");
+        PGSQL::IgnoreNotice        = Config::GetBool(ini,  pgsql, "IgnoreNotice");
+        PGSQL::LogNotice           = Config::GetBool(ini,  pgsql, "LogNotice");
 
     }
 

